src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs

```diff
@@ -4,26 +4,31 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using Microsoft.OData.Edm.Vocabularies;
-
 namespace Microsoft.OData.Edm.Csdl.Parsing.Ast
 {
     /// <summary>
-    /// Represents a CSDL function return type.
+    /// Represents a CSDL operation return type.
     /// </summary>
-    internal class CsdlOperationReturnType : CsdlElement, IEdmVocabularyAnnotatable
+    internal class CsdlOperationReturnType : CsdlElement
     {
-        private readonly CsdlTypeReference returnType;
-
         public CsdlOperationReturnType(CsdlTypeReference returnType, CsdlLocation location)
             : base(location)
         {
-            this.returnType = returnType;
+            this.ReturnType = returnType;
+        }
+
+        public CsdlTypeReference ReturnType { get; }
+
+        public override void AddAnnotation(CsdlDirectValueAnnotation annotation)
+        {
+            base.AddAnnotation(annotation);
+            ReturnType.AddAnnotation(annotation);
         }
 
-        public CsdlTypeReference ReturnType
+        public override void AddAnnotation(CsdlAnnotation annotation)
         {
-            get { return this.returnType; }
+            base.AddAnnotation(annotation);
+            ReturnType.AddAnnotation(annotation);

```

Comment:
As above; should we be annotating *both* the CsdlTypeReference and the CsdlOperationReturnType?

Reply:
Basically, we should annotate "CsdlOperationReturnType" only, however, we use the "CsdlTypeReference returnType" to constructor the other ""CsdlSemanticsOperation....ReturnType"'. 

So, I have to do like this. Of course, we can use "base.AddAnnotation(annotation), but for safety, it 's better to keep the base call.